### DISCUSSION TITLE
How to find type of project?

### DISCUSSION THREAD

**Discussion by soroshsabz** [cake-build/cake#3122] [Q&A]
ITNOA

I use [cake build system](https://cakebuild.net/) to automate building my solution and I want to automate testing my project with cake in our CI,

My NUnit projects have many types such as asp.net dot net framework 4.6.1 or dot net core 2.1 or ...

my problem is when I want to run our unit test with [NUnit3 cake dsl](https://cakebuild.net/dsl/nunit-v3/#Built-In) we need to know dll path of each project, so I need to know what is type of each project, because for example for dot net core 2.1 dll is under netcoreapp directory and for another project dll is under somewhere else.

for better demonstration please see below image

[![Structure of my solution][1]][1]

As you can see in Test Solution Folder we have many test projects with different type framework and ... (for example one of them is `dotnetcore2.1` another one is `dot net framework 4.6.1` and etc.)

So my question is how to find type of project?

Is there any solution to [NUnit3](https://cakebuild.net/dsl/nunit-v3/#Built-In) found dll by self?

[my Stack overflow question](https://stackoverflow.com/q/66172925/1539100)

thanks

  [1]: https://i.stack.imgur.com/AdG6M.png

**Comment by pascalberger** [comment]
This has been cross-posted to [Stackoverflow](https://stackoverflow.com/q/66172925/1539100) and answers there

**Reply by soroshsabz** [reply]
As you can see in my question that I linked to it in last line, Any of answers exist in stackoverflow does not resolve my problem, I write some comment in each answer to describe my problem better.

**Comment by augustoproiete** [comment]
@soroshsabz You could try to use the [`GetFiles`](https://cakebuild.net/api/Cake.Common.IO/GlobbingAliases/BCA1F865) alias to find all the test assemblies, and then inspect the `FullPath` of each assembly to check the `TFM`, and decide what to do then...

```csharp
var files = GetFiles("./**/*.Tests.dll");
foreach(var file in files)
{
    Information("File: {0}", file.FullPath);

    if (file.FullPath.Contains("/net461/")
    {
        // ... Assembly is for .NET 4.6.1
    }
    else if (file.FullPath.Contains("/netcoreapp2.1/")
    {
        // ... Assembly is for .NET Core 2.1
    }
    // etc...
}
```
